presto-hive/src/main/java/com/facebook/presto/hive/S3SelectRecordCursor.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.spi.type.TypeManager;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.RecordReader;
+import org.joda.time.DateTimeZone;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.apache.hadoop.hive.serde.serdeConstants.LIST_COLUMNS;
+import static org.apache.hadoop.hive.serde.serdeConstants.LIST_COLUMN_TYPES;
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_DDL;
+
+class S3SelectRecordCursor<K, V extends Writable>
+        extends GenericHiveRecordCursor
+{
+    private static final String THRIFT_STRUCT = "struct";
+    private static final String START_STRUCT = "{";
+    private static final String END_STRUCT = "}";
+    private static final String FIELD_SEPARATOR = ",";
+    private static final String EMPTY_STRING = "";
+    private static final String SPACE = " ";
+
+    S3SelectRecordCursor(
```

Comment:
Make this public, since the constructor is logically part of the public API for the class. Access is controlled at the class level. If we wanted to later make the class public, we wouldn't need to change the access of the constructor.